src/wallet/rpcwallet.cpp

```diff
@@ -3007,6 +3011,111 @@ UniValue z_getnewaddress(const UniValue& params, bool fHelp)
+            "z_getnewaccount\n"
+            "\nSelects the next available sequential account number within the UA-compatible HD seed phrase;"
+            "\ngenerates the corresponding spend authorities for the best and second-best shielded pools, and the transparent pool;"
+            "\nreturns the new account number and the default / first UA containing the default addresses for each pool.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"account\": n,       (numeric) next available account number\n"
+            "  \"unifiedaddress\"    (string) The default address for this account\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("z_getnewaccount", "")
+            + HelpExampleRpc("z_getnewaccount", "")
+        );
+
+    if (!fExperimentalOrchardWallet) {
+        throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Error: the Orchard wallet experimental extensions are disabled.");
+    }
+    int64_t account = 999999; // TODO placeholder
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("account", account);
+    result.pushKV("unifiedaddress", "TODO");
+    return result;
+}
+
+UniValue z_getaddressforaccount(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+    if (fHelp || params.size() < 1 || params.size() > 3)
+        throw runtime_error(
+            "z_getaddressforaccount account ( diversifier_index [\"pool\", ...] )\n"
+            "\nFor the given account number, return a UA in accordance with the remaining arguments."
+            "\nIf no diversifier index is given (or the string \"*\"), the next unused index will be selected."
+            "\nCompute a Unified Address corresponding to the diversifier index, containing addresses"
+            "\nfor the given pools (current default [\"transparent\",\"sapling\",\"orchard\"]).\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"account\": n,                    (numeric) the specified account number\n"
+            "  \"diversifier\": n,                (numeric) the index specified or chosen\n"
+            "  \"pools\": [\"pool\",...]\",         (json array of string) the pools (e.g. \"transparent\", \"orchard\") the UA contains\n"
+            "  \"unifiedaddress\"                 (string) The corresponding address\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("z_getaddressforaccount", "4")
+            + HelpExampleCli("z_getaddressforaccount", "4 1")
+            + HelpExampleCli("z_getaddressforaccount", "4 1 '[\"transparent\",\"sapling\",\"orchard\"]'")
+            + HelpExampleRpc("z_getaddressforaccount", "4")
+        );
+
+    if (!fExperimentalOrchardWallet) {
+        throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Error: the Orchard wallet experimental extensions are disabled.");
+    }
+    int64_t account = params[0].get_int64();
+    if (account < 0 || account > 0x7fffffff-1) {
```

Comment:
I believe the current code is as-intended. The range is [0, 0x7FFFFFFF); 0x7FFFFFFE is allowed.

Reply:
Yep, you're correct. I was for some reason reading this as if the constant was `0x80000000-1`. It would be clearer if we had `LEGACY_ADDRESSES_ACCOUNT = 0x7fffffff` and then this was:
```suggestion
    if (account < 0 || account >= LEGACY_ADDRESSES_ACCOUNT) {
```